src/main/java/ro/msg/learning/shop/services/MyUserDetailService.java

```diff
@@ -0,0 +1,54 @@
+package ro.msg.learning.shop.services;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.exceptions.RecordNotFoundException;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.repository.CustomerRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@AllArgsConstructor
+@Service
+public class MyUserDetailService implements UserDetailsService {
+    private CustomerRepository customerRepository;
+
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+        Optional<Customer> customer = customerRepository.findCustomerByuserName(s);
```

Comment:
Instead of using `isPresent` and then `get`, you could use `map()` and then `orElseThrow`.